src/Compilers/Core/AnalyzerDriver/DeclarationComputer.cs

```diff
@@ -13,9 +13,7 @@ internal class DeclarationComputer
         internal static DeclarationInfo GetDeclarationInfo(SemanticModel model, SyntaxNode node, bool getSymbol, IEnumerable<SyntaxNode> executableCodeBlocks, CancellationToken cancellationToken)
         {
             var declaredSymbol = getSymbol ? model.GetDeclaredSymbol(node, cancellationToken) : null;
-            var codeBlocks = executableCodeBlocks == null ?
-                ImmutableArray<SyntaxNode>.Empty :
-                executableCodeBlocks.Where(c => c != null).AsImmutableOrEmpty();
+            var codeBlocks = executableCodeBlocks?.Where(c => c != null).AsImmutableOrEmpty() ?? ImmutableArray<SyntaxNode>.Empty;
```

Comment:
To be clear: I don't have any objections to this pattern in general.  It definitely has concrete value in cases where it removes an extra side effect.

For readability it's more murky.  I toyed with some samples on my box after writing my comments.  In some cases I thought the resulting code looked better, in others I thought it looked worse.  

I don't want my comment to be viewed as blocking, just a thought I had while reading it.